packages/network-core/src/Request.ts

```diff
@@ -0,0 +1,72 @@
+import { PureModel, PureCollection, getModelType } from '@datx/core';
+import { IResponseObject } from '.';
+import { IAsync } from './interfaces/IAsync';
+import { IGeneralize } from './interfaces/IGeneralize';
+import { INetwork } from './interfaces/INetwork';
+import { IRefs } from './interfaces/IRefs';
+import { IRequestDetails } from './interfaces/IRequestDetails';
+import { IResponseSnapshot } from './interfaces/IResponseSnapshot';
+import { ISubrequest } from './interfaces/ISubrequest';
+import { Response } from './Response';
+
+export class Request<
+  TNetwork extends INetwork,
+  TModelClass extends typeof PureModel = typeof PureModel,
+  TResponse extends TModelInstance | Array<TModelInstance | unknown> | unknown =
+    | InstanceType<TModelClass>
+    | Array<InstanceType<TModelClass>>,
+  TModelInstance extends
+    | (InstanceType<TModelClass> & PureModel)
+    | unknown = InstanceType<TModelClass>,
+  // @ts-ignore No way to avoid this :( But the final type is correct
+  IA extends IAsync<TModelInstance> = ReturnType<TNetwork['execAll']>,
+> {
+  constructor(
+    protected readonly refs: IRefs<TNetwork, typeof Request>,
+    protected readonly requestData: IRequestDetails,
+    protected readonly subrequests: Array<ISubrequest<TResponse, TNetwork, typeof Request>> = [],
+  ) {}
+
+  public fetch(
```

Comment:
@isBatak as long as the method signature stays the same, you're free to override this method in any way you want in `SwrRequest` or `MyProjectRequest`